src/bse_print.F

```diff
@@ -0,0 +1,820 @@
+      REAL(KIND=dp)                                      :: d_eh_dir, d_exc_dir, sigma_e_dir, &
+                                                            sigma_h_dir
+      TYPE(particle_type), DIMENSION(:), POINTER         :: particle_set
+
+      CALL timeset(routineN, handle)
+      CALL get_qs_env(qs_env, particle_set=particle_set)
+      IF (unit_nr > 0) THEN
+         WRITE (unit_nr, '(T2,A4,T7,A)') 'BSE|', &
+            'Exciton descriptors for excitation level n are given by'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T15,A)') 'BSE|', &
+            'd_eh   = | <r_h - r_e>_exc |'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T15,A)') 'BSE|', &
+            'σ_e    = sqrt( <r_e^2>_exc - <r_e>_exc^2 )'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T15,A)') 'BSE|', &
+            'σ_h    = sqrt( <r_h^2>_exc - <r_h>_exc^2 )'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T15,A)') 'BSE|', &
+            'COV_eh = <r_e r_h>_exc - <r_e>_exc <r_h>_exc'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T15,A)') 'BSE|', &
+            'd_exc  = sqrt( | < |r_h - r_e|^2 >_exc )'
+         WRITE (unit_nr, '(T2,A4,T15,A)') 'BSE|', &
+            '       = sqrt( d_eh^2 + σ_e^2 + σ_h^2 - 2 * COV_eh )'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T15,A)') 'BSE|', &
+            'R_eh   = COV_eh / (σ_e * σ_h)'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T7,A)') 'BSE|', &
+            'where the expectation values <.>_exc are taken with respect to the '
+         WRITE (unit_nr, '(T2,A4,T7,A)') 'BSE|', &
+            'exciton wavefunction  of excitation n:'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+
+         IF (exc_descr(1)%flag_TDA) THEN
+            WRITE (unit_nr, '(T2,A4,T20,A)') 'BSE|', &
+               '𝚿_n(r_e,r_h) = Σ_{i,a} X_ia^n ψ_i(r_h) ψ_a(r_e)  ,'
+         ELSE
+            WRITE (unit_nr, '(T2,A4,T20,A)') 'BSE|', &
+               '𝚿_n(r_e,r_h) = Σ_{i,a} X_ia^n ψ_i(r_h) ψ_a(r_e)'
+            WRITE (unit_nr, '(T2,A4,T40,A)') 'BSE|', &
+               '+ Y_ia^n ψ_a(r_h) ψ_i(r_e) ,'
+         END IF
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T7,A)') 'BSE|', &
+            'i.e.'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         WRITE (unit_nr, '(T2,A4,T20,A)') 'BSE|', &
+            '< O >_exc = < 𝚿_n | O | 𝚿_n > / < 𝚿_n | 𝚿_n >  ,'
+         WRITE (unit_nr, '(T2,A4)') 'BSE|'
+         IF (exc_descr(1)%flag_TDA) THEN
+            WRITE (unit_nr, '(T2,A4,T7,A)') 'BSE|', &
+               'where c_n = < 𝚿_n | 𝚿_n > = 1 within TDA.'
+         ELSE
+            WRITE (unit_nr, '(T2,A4,T7,A)') 'BSE|', &
+               'where c_n = < 𝚿_n | 𝚿_n > deviates from 1 without TDA.'
+         END IF
+
+         ! WRITE (unit_nr, '(T2,A4)') 'BSE|'
```

Comment:
Are these WRITE statements still needed? Remove if not or wrap in `IF (.FALSE.) THEN ... END IF` for type checking.